src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -173,12 +174,31 @@ - (instancetype)initWithFactory:(MTRDeviceControllerFactory *)factory
                 return nil;
             }
 
+            id<MTRDeviceControllerStorageDelegate> storageDelegateToUse = storageDelegate;
+#if MTR_PER_CONTROLLER_STORAGE_ENABLED
```

Comment:
This could be moved out of the real code by exposing `_controllerDataStore` as writable to tests, or factoring this logic out into a internal `createControllerDataStore` method that the test can override by sub-classing MTRDeviceController.